### DISCUSSION TITLE
How can I prevent the default roles from reappearing when I update teleport

### DISCUSSION THREAD

**Discussion by tooboredtocode** [gravitational/teleport#6766] [Q&A]
I like to delete the builtin roles, in favour of my own, to gain more control about the permissions.

However, every time I update Teleport the default roles keep appearing again. Is there a way to prevent this from happening?

**Comment by webvictim** [comment]
That's odd behaviour, I wouldn't have expected that Teleport would re-add deleted roles. How about if you just edit the roles and remove all their permissions (basically move everything to the `deny` section rather than `allow`) - do the permissions get changed back when you update?

**Reply by tooboredtocode** [reply]
When I change the permissions they remain, just that the deleted roles reappear each time I restart the auth server.
